File: solutions/scala/bob/1/src/main/scala/Bob.scala
object Bob {
  def response(statement: String): String = {
    val questionMark = statement.strip.endsWith("?")
    val yell = statement.exists(_.isLetter) && statement.forall(c => !c.isLetter || c.isUpper)
    val silence = statement.forall(c => c.isWhitespace)
    if(silence) "Fine. Be that way!"
    else if(yell && questionMark) "Calm down, I know what I'm doing!"
    else if(yell) "Whoa, chill out!"
    else if(questionMark) "Sure."
    else "Whatever."
  }
}

File: solutions/python/bob/1/bob.py
def response(hey_bob):
    ends_question_mark = hey_bob.strip().endswith("?")
    all_capital_letters = hey_bob.isupper()
    
    if ends_question_mark and all_capital_letters:
        return "Calm down, I know what I'm doing!"
    elif ends_question_mark:
        return "Sure."
    elif all_capital_letters:
        return "Whoa, chill out!"
    elif hey_bob.isspace() or len(hey_bob) == 0:
        return "Fine. Be that way!"
    else:
        return "Whatever."
